### DISCUSSION TITLE
Cannot install anything from homebrew.bintray.com; get curl error.

### DISCUSSION THREAD

**Discussion by eric-b-andrews** [Homebrew/discussions#1248] [Everyday usage] [answered]
As per title, any attempts to install packages from homebrew.bintray.com give the following error:

`curl: (60) SSL: no alternative certificate subject name matches target host name 'homebrew.bintray.com'`

I can get packages from other hosts (e.g., Adobe Reader installs just fine), but not bintray.

If I try to go to https://homebrew.bintray.com/bottles, I get a certificate error; it lists the certificate name as `*.us-west-2.es.amazonaws.com`, but on a working machine it lists the certificate name as `*.bintray.com`.

I've tried reinstalling homebrew, using a different network, and gone through a whole call with Apple support to no avail. The only warning I get with `brew doctor` is that `guile@2` is outdated.

Any help is appreciated!

**Comment by carlocab** [comment, marked as answer]
What's the output of `brew config`?

**Reply by eric-b-andrews** [reply]
HOMEBREW_VERSION: 3.0.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8e11f03fa93b148e26044a331654241a2c307039
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f7316068d9ad4f83f79992beee84c294ea39e9c7
Core tap last commit: 38 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.JkZL6IJKjZ/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: N/A
Xcode: 12.4
XQuartz: 2.7.11 => /opt/X11

**Reply by carlocab** [reply]
What about
```
curl -LIv https://homebrew.bintray.com
```
?

**Reply by eric-b-andrews** [reply]
*   Trying 54.188.157.32...
* TCP_NODELAY set
* Connected to homebrew.bintray.com (54.188.157.32) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.us-west-2.es.amazonaws.com
*  start date: May  5 00:00:00 2020 GMT
*  expire date: Jun  5 12:00:00 2021 GMT
*  subjectAltName does not match homebrew.bintray.com
* SSL: no alternative certificate subject name matches target host name 'homebrew.bintray.com'
* Closing connection 0
* TLSv1.2 (OUT), TLS alert, close notify (256):
curl: (60) SSL: no alternative certificate subject name matches target host name 'homebrew.bintray.com'
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

**Reply by carlocab** [reply]
It looks like you're being connected elsewhere for some reason. It sounds more like a network-related issue. Are you running a firewall/packet analyser/etc (e.g. Little Snitch)? You might also have a weird DNS setup.

Failing figuring that out, this might work:
```
HOMEBREW_BOTTLE_DOMAIN=https://ghcr.io/v2/Homebrew/core brew install <formula>
```
It's likely that this only works if you're on the development version of `brew`, though. To switch to it,
```
HOMEBREW_DEVELOPER=1 brew update
```
I'd recommend this only as a last resort, though, since switching back to the stable version of `brew` could get complicated.

**Reply by eric-b-andrews** [reply]
The bottle domain switch works! Thank you! I did have to switch to the development version, but if that causes problems down the line I guess I'll just have to solve them as they pop up.

As far as the network side of things is concerned, I'm not running any firewall except the built-in macOS firewall, and turning that off didn't change anything. As far as I know my DNS setup is pretty normal—I haven't done anything to muck with it, but I also don't know a whole lot about networking.

**Reply by carlocab** [reply]
If you need to switch back, I _think_ this should be enough:
```
git -C $(brew --repository) checkout <latest stable tag>
```
though you might also need to delete `devcmdrun = true` from `$(brew --repository)/.git/config` (in case it's there).

You may also want to do `brew update-reset` after that for good measure, but please make a note of this:
```
❯ brew help update-reset
Usage: brew update-reset [repository ...]

Fetch and reset Homebrew and all tap repositories (or any specified
repository) using git(1) to their latest origin/HEAD.

Note: this will destroy all your uncommitted or committed changes.
```
While you're on the dev version you shouldn't need to set `HOMEBREW_BOTTLE_DOMAIN`. The bottle domain showed above should already be the default. This will also be the new bottle domain in the next tagged release of `brew`.

**Reply by eric-b-andrews** [reply]
Thank you for all of your help! I really appreciate it.
